packages/integration-tests/src/tests/ui/applications/helpers.ts

```diff
@@ -0,0 +1,136 @@
+import { appendPath } from '@silverhand/essentials';
+import { type Page } from 'puppeteer';
+
+import {
+  expectToClickDetailsPageOption,
+  expectToOpenNewPage,
+  waitForToast,
+} from '#src/ui-helpers/index.js';
+import { expectNavigation } from '#src/utils.js';
+
+import { type ApplicationCase } from './application-test-cases.js';
+
+export const expectToClickFramework = async (page: Page, framework: string) => {
+  const frameworkCard = await expect(page).toMatchElement(
+    'div[class*=card]:has(div[class$=header] div[class$=name])',
+    {
+      text: framework,
+    }
+  );
+
+  await expect(frameworkCard).toClick('button span', { text: 'Start Building' });
+};
+
+export const expectFrameworkExists = async (page: Page, framework: string) => {
+  await expect(page).toMatchElement('div[class*=card]:has(div[class$=header] div[class$=name])', {
+    text: framework,
+  });
+};
+
+export const expectToProceedCreationFrom = async (
+  page: Page,
+  { name, description }: ApplicationCase
+) => {
+  // Expect the creation form to be open
+  await expect(page).toMatchElement(
+    '.ReactModalPortal div[class$=header] div[class$=titleEllipsis]',
+    {
+      text: 'Create Application',
+    }
+  );
+
+  await expect(page).toFillForm('form', {
```

Comment:
Should we narrow down the form selector within the CreateApplication Modal?

Reply:
Updated.

Reply:
Updated.